pkg/linter/linter.go

```diff
@@ -593,6 +598,52 @@ func pkgconfTestLinter(_ context.Context, cfg *config.Configuration, pkgname, pa
 	return fmt.Errorf("pkgconfig directory found")
 }
 
+var isSharedObjectFileRegex = regexp.MustCompile(`\.so$`)
+
+func lddcheckTestLinter(_ context.Context, cfg *config.Configuration, pkgname, path string) error {
+	if !isSharedObjectFileRegex.MatchString(path) {
+		return nil
+	}
+
+	if cfg == nil {
+		return fmt.Errorf("shared object found and missing .melange.yaml")
+	}
+
+	if cfg.Package.Name == pkgname {
+		if cfg.Test != nil {
+			for _, test := range cfg.Test.Pipeline {
+				if test.Uses == "test/ldd-check" {
+					return nil
+				} else if test.Uses == "test/tw/ldd-check" {
+					return nil
+				}
+			}
+		}
+	} else {
+		for _, p := range cfg.Subpackages {
+			if p.Name != pkgname {
```

Comment:
We really should rename `pkgname` to `testname` (here and elsewhere, but no need to do it in this PR) in order to avoid that problem we were investigating :-/.